src/libraries/Common/src/Extensions/ParameterDefaultValue/ParameterDefaultValue.netcoreapp.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Extensions.Internal
+{
+    internal static class ParameterDefaultValue
+    {
+        public static bool TryGetDefaultValue(ParameterInfo parameter, out object? defaultValue)
+        {
+            bool tryToGetDefaultValue = true;
+            defaultValue = null;
+            bool hasDefaultValue = parameter.HasDefaultValue;
+
+            if (hasDefaultValue)
+            {
+                if (tryToGetDefaultValue)
+                {
+                    defaultValue = parameter.DefaultValue;
```

Comment:
Can we just refactor this one line (i.e. just the `try-catch`) into a partial class? That way we don't need to duplicate the rest of the method?

Reply:
I guess this comment really belongs up on line 19:

```C#
bool hasDefaultValue = parameter.HasDefaultValue;
```

that is the line that has a try-catch around it.